diff --git a/tclconfig/tcl.m4 b/tclconfig/tcl.m4
index 57489c9..c6af495 100644
--- a/tclconfig/tcl.m4
+++ b/tclconfig/tcl.m4
@@ -1646,7 +1646,7 @@ dnl AC_CHECK_TOOL(AR, ar)
 	    # Version numbers are dot-stripped by system policy.
 	    TCL_TRIM_DOTS=`echo ${VERSION} | tr -d .`
 	    UNSHARED_LIB_SUFFIX='${TCL_TRIM_DOTS}.a'
-	    SHARED_LIB_SUFFIX='${TCL_TRIM_DOTS}\$\{DBGX\}.so.1'
+	    SHARED_LIB_SUFFIX='${TCL_TRIM_DOTS}\$\{DBGX\}.so'
 	    TCL_LIB_VERSIONS_OK=nodots
 	    ;;
 	Darwin-*)
@@ -3234,6 +3234,7 @@ print("manifest needed")
 	RANLIB_STUB=:
     else
 	RANLIB_STUB="${RANLIB}"
+	TCL_TRIM_DOTS='`echo ${PACKAGE_VERSION} | tr -d .`'
 	if test "${SHARED_BUILD}" = "1" ; then
 	    SHLIB_LD_LIBS="${SHLIB_LD_LIBS} ${TCL_STUB_LIB_SPEC}"
 	    if test x"${TK_BIN_DIR}" != x ; then
