diff --git a/configure.ac b/configure.ac
index 4b6ef57..a5e28e3 100644
--- a/configure.ac
+++ b/configure.ac
@@ -50,6 +50,8 @@ AC_SUBST(PAPERSIZEVAR)
 AH_TEMPLATE(PAPERSIZEVAR, [Environment variable that overrides the default paper size])
 AC_DEFINE_UNQUOTED(PAPERSIZEVAR, "$PAPERSIZEVAR")
 
+AC_CHECK_HEADERS([malloc.h])
+
 dnl Create files
 dnl ============
 AC_OUTPUT([
diff --git a/lib/paper.c b/lib/paper.c
index 1b25492..010ed79 100644
--- a/lib/paper.c
+++ b/lib/paper.c
@@ -14,7 +14,9 @@
 #include <sys/stat.h>
 
 #include <stdio.h>
+#ifdef HAVE_MALLOC_H
 #include <malloc.h>
+#endif
 #include <string.h>
 #include <ctype.h>
 
